src/dcp/dcp.c

```diff
@@ -442,77 +445,136 @@ int main(int argc, char** argv)
+    else {
+        daos_epoch_t epoch;
+        if (rank == 0) {
+            rc = daos_obj_list_oids(daos_args, &epoch, flist);
+        }
+	    if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "DAOS failed to list oids: ", MFU_ERRF,
+                    MFU_ERRP(-MFU_ERR_DAOS));
+            rc = 1;
+        }
+
+        /* evenly spread obj ids among each rank */
+        mfu_flist_summarize(flist);
+        mfu_flist newlist = mfu_flist_spread(flist);
+
+        /* perform copy after oids are spread evenly across all ranks */
+        flist_t* local_flist = (flist_t*) newlist;
+        rc = daos_obj_copy(daos_args, local_flist);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "DAOS object copy failed: ", MFU_ERRF,
+                    MFU_ERRP(-MFU_ERR_DAOS));
+            rc = 1;
+        }
+
+        /* wait here until until all procs are done copying,
+         * then destroy snapshot */
+        MPI_Barrier(MPI_COMM_WORLD);
 
-    if (inputname == NULL) {
-        /* if daos is set to SRC then use daos_ functions on walk */
-        mfu_flist_walk_param_paths(numpaths_src, paths, walk_opts, flist, mfu_src_file);
-    } else {
-        struct mfu_flist_skip_args skip_args;
-
-        /* otherwise, read list of files from input, but then stat each one */
-        mfu_flist input_flist = mfu_flist_new();
-        mfu_flist_read_cache(inputname, input_flist);
-
-        skip_args.numpaths = numpaths_src;
-        skip_args.paths = paths;
-        mfu_flist_stat(input_flist, flist, input_flist_skip, (void *)&skip_args,
-                       walk_opts->dereference, mfu_src_file);
-        mfu_flist_free(&input_flist);
+        /* destroy snapshot after copy */
+        if (rank == 0) {
+            daos_epoch_range_t epr;
+            epr.epr_lo = epoch;
+            epr.epr_hi = epoch;
+            rc = daos_cont_destroy_snap(daos_args->src_coh, epr, NULL);
+            if (rc != 0) {
+                MFU_LOG(MFU_LOG_ERR, "DAOS destroy snapshot failed: ", MFU_ERRF,
+                        MFU_ERRP(-MFU_ERR_DAOS));
+                rc = 1;
+            }
+        }
+
+	    /* free newlist that was created for non-posix copy */
+	    mfu_flist_free(&newlist);
     }
 
-    /* copy flist into destination */ 
-    rc = mfu_flist_copy(flist, numpaths_src, paths,
-                            destpath, mfu_copy_opts, mfu_src_file,
-                            mfu_dst_file);
-    if (rc < 0) {
-        /* hit some sort of error during copy */
+    // Synchronize rc across all processes
+    if (!mfu_alltrue(rc == 0, MPI_COMM_WORLD)) {
         rc = 1;
     }
 
-    /* free the file list */
-    mfu_flist_free(&flist);
-
-    /* free the path parameters */
-    mfu_param_path_free_all(numpaths, paths);
-
-    /* free memory allocated to hold params */
-    mfu_free(&paths);
+    // Rank 0 prints success if needed
+    if (rc == 0 && rank == 0 && !is_posix_copy) {
+        printf("Successfully copied to DAOS Destination Container.\n");
```

Comment:
Just converting to MFU_LOG and dropping the newline.
```suggestion
        MFU_LOG(MFU_LOG_INFO, "Successfully copied to DAOS Destination Container.");
```